internal/controller/management/upgrade/serviceaccounts.go

```diff
@@ -0,0 +1,98 @@
+package upgrade
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	rbacapi "github.com/akuity/kargo/api/rbac/v1alpha1"
+	"github.com/akuity/kargo/internal/logging"
+)
+
+// Old annotation keys
+const (
+	OldAnnotationKeySub    = "rbac.kargo.akuity.io/sub"
+	OldAnnotationKeyEmail  = "rbac.kargo.akuity.io/email"
+	OldAnnotationKeyGroups = "rbac.kargo.akuity.io/groups"
```

Comment:
Nit: There's no reason to export these. This comment is non-blocking. It is up to you whether it's worth addressing.